modules/kubernetes-addons/aws-coredns/main.tf

```diff
@@ -11,3 +24,45 @@ resource "aws_eks_addon" "coredns" {
     try(var.addon_config.tags, {})
   )
 }
+
+module "helm_addon" {
+  source = "../helm-addon"
+  count  = var.use_managed_addon ? 0 : 1
```

Comment:
From an addon perspective, they are fulfilling a similar purpose: CoreDNS. The difference is - is it managed by the EKS service or by the user via a helm chart. 

I would like to avoid the expansion of modules because they come with a lot of overhead and they aren't really living up to modules. But, I would love to hear more on why this is not a good approach